src/charm.py

```diff
@@ -111,14 +102,44 @@ def generate_worker_layer(self, worker: Worker) -> Layer:
             }
         )
 
+    def _add_juju_topolgy(self):
+        """Modify the config file to add juju topology for
+        `metrics-generator` metrics before running the pebble service.
+        """
+        tempo_container = self.unit.get_container("tempo")
+        if not tempo_container.exists(CONFIG_FILE):
+            logger.debug("Skipping adding juju topology. Config file does not exist yet.")
+            return
+
+        raw_data = tempo_container.pull(CONFIG_FILE).read()
+        data = yaml.safe_load(raw_data)
+        # if `metrics_generator` doesn't exist in config,
+        # then it is not enabled, so no point of adding juju topology.
+        if "metrics_generator" in data:
+            if "registry" not in data["metrics_generator"]:
+                data["metrics_generator"]["registry"] = {}
+            data["metrics_generator"]["registry"]["external_labels"] = dict(
+                self.worker.cluster.juju_topology.as_dict()
+            )
+
+            # write new data
+            tempo_container.push(CONFIG_FILE, yaml.safe_dump(data))
+
     def _on_collect_status(self, e: CollectStatusEvent):
-        # add Tempo worker custom blocking conditions
-        if self.worker.roles and self.worker.roles[0] not in self._valid_roles:
-            e.add_status(
-                BlockedStatus(
-                    f"Invalid `role` config value: {self.config.get('role')!r}. Should be one of {self._valid_roles}"
+        # add Tempo worker-specific statuses
+        if (
+            self.worker.cluster.relation
+            and self.worker.roles
+            and not self.worker.cluster.get_remote_write_endpoints()
+        ):
+            if self.worker.roles[0] == "all":
+                e.add_status(
+                    ActiveStatus(
+                        "metrics-generator disabled. No prometheus remote-write relation configured"
+                    )
                 )
-            )
+            elif self.worker.roles[0] == "metrics-generator":
+                e.add_status(BlockedStatus("No prometheus remote-write relation configured"))
```

Comment:
```suggestion
                e.add_status(BlockedStatus("No prometheus remote-write relation configured. Please relate {self.worker.cluster.app.name} to a remote-write endpoint."))
```

Reply:
(or something similar)

Reply:
I think the line is too long it will truncate in `juju status`? 
I'd make it "No prometheus remote-write relation configured on the coordinator"